modules/geo-layers/src/dggs-layer/dggs-decoders/h3-decoder.ts

```diff
@@ -0,0 +1,122 @@
+// deck.gl
+// SPDX-License-Identifier: MIT
+// Copyright (c) vis.gl contributors
+
+// import {type Bounds2D} from '@math.gl/types';
+import {type DGGSDecoder} from './dggs-decoder';
+import {
+  type CoordPair,
+  type H3IndexInput, // Either string or SplitLong
+  type SplitLong,
+  cellToBoundary,
+  cellToLatLng,
+  h3IndexToSplitLong,
+  cellsToMultiPolygon
+} from 'h3-js';
+import {lerp} from '@math.gl/core';
+
+/**
+ * "Standardized" API for working with A5 DGGS tokens.
+ * @note Copy of same type in math.gl. They are structurally identical. Will be replaced by an import in the future.
+ */
+export const H3Decoder = {
```

Comment:
Wonder if the way to go here is to make the decoder generic on "native" representation of the index type. So for example H3 it would be a `string`, while for A5 a `bigint`. This way unnecessary conversions could be avoided